cli/dcoscli/main.py

```diff
@@ -60,8 +60,9 @@ def _main():
     if dcoscli.version != 'SNAPSHOT' and command and command != "config":
         try:
             cluster_id = mesos.DCOSClient().metadata().get('CLUSTER_ID')
-        except DCOSException:
-            raise
+        except DCOSException as e:
+            if "Authentication failed" in str(e):
```

Comment:
no, we want to throw the 401 (Authenticated failed) error, but not the 403 error, because that could be valid in the user specified command. I'll add `DCOSAuthException` so that the code is more readable